tests/Unit/NumericalAlgorithms/LinearOperators/Test_CoefficientTransforms.cpp

```diff
@@ -0,0 +1,162 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/IndexIterator.hpp"
+#include "DataStructures/ModalVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Mesh.hpp"
+#include "NumericalAlgorithms/LinearOperators/CoefficientTransforms.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+
+namespace {
+namespace test_legendre_polynomials {
+void test_1d() noexcept {
+  for (size_t order = 1; order < 7; ++order) {
+    CAPTURE(order);
+    const Mesh<1> mesh(order + 1, Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto);
+    const DataVector& x = Spectral::collocation_points(mesh);
+    const auto helper = [&mesh, order, &x ](const size_t second_term) noexcept {
+      DataVector u_expected =
+          Spectral::compute_basis_function_value<Spectral::Basis::Legendre>(
+              order, x);
```

Comment:
Could you add comments to these tests please?

Reply:
Sure, but could you request something more specific? I'm not sure what you think needs clarification.